packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/editingcontext/EditingContextSearchService.java

```diff
@@ -107,10 +119,17 @@ public Optional<IEditingContext> findById(String editingContextId) {
 
         this.logger.debug("{} documents loaded for the editing context {}", resourceSet.getResources().size(), editingContextId);
 
+        Map<String, IRepresentationDescription> representationDescriptions = new LinkedHashMap<>();
+        var registry = new RepresentationDescriptionRegistry();
+        this.configurers.forEach(configurer -> configurer.addRepresentationDescriptions(registry));
+        registry.getRepresentationDescriptions().forEach(representationDescription -> representationDescriptions.put(representationDescription.getId(), representationDescription));
+        this.dynamicRepresentationDescriptionService.findDynamicRepresentationDescriptions(editingContextId, editingDomain)
+                .forEach(representationDescription -> representationDescriptions.put(representationDescription.getId(), representationDescription));
+
         long end = System.currentTimeMillis();
         this.timer.record(end - start, TimeUnit.MILLISECONDS);
 
-        return Optional.of(new EditingContext(editingContextId, editingDomain));
+        return Optional.of(new EditingContext(editingContextId, editingDomain, representationDescriptions));
```

Comment:
Is the `IEditingContext` used by some query/mutation/subscription without the need of representation descriptions? If it is the case, may be the computation of existing representation descriptions could be computed in  implementations of `IRepresentationDescriptionSearchService` ?